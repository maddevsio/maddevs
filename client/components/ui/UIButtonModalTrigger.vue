<template>
  <button
    v-WaveAnimation
    :class="{
      'ui-button-modal-trigger--red': isRed,
      'ui-button-modal-trigger--black': isBlack,
      'ui-button-modal-trigger--grey': isGrey,
    }"
    class="ui-button-modal-trigger"
    @click="showModal"
  >
    {{ buttonInnerText }}
  </button>
</template>

<script>
import WaveAnimation from '@/directives/WaveAnimation'

export default {
  name: 'UIButtonModalTrigger',
  directives: {
    WaveAnimation,
  },

  props: {
    buttonInnerText: {
      type: String,
      default: 'Button Inner Text',
    },

    isRed: {
      type: Boolean,
      default: false,
    },

    isBlack: {
      type: Boolean,
      default: false,
    },

    isGrey: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    showModal() {
      this.$emit('onClick')
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.ui-button-modal-trigger {
  height: 65px;
  border: 1px solid $border-color--grey;
  border-radius: 2px;
  box-shadow: none;
  background-color: transparent;
  font-size: 18px;
  font-family: 'Poppins-Regular', sans-serif;
  cursor: pointer;
  transition: 0.4s;

  &--red {
    color: $text-color--red;
    border: 1px solid $border-color--red;

    &:hover {
      background-color: $bgcolor--red;
      color: $text-color--white;
    }
  }

  &--black {
    color: $text-color--black;
    border: 1px solid $border-color--black;

    &:hover {
      background-color: $bgcolor--black;
      color: $text-color--white;
    }
  }

  &--grey {
    color: $text-color--grey;
    border: 1px solid $border-color--grey;

    &:hover {
      background-color: $bgcolor--grey;
      color: $text-color--black;
    }
  }

  &:active {
    background-color: $button-active--red;
    border-color: $button-active-border--red;
  }
}
</style>
