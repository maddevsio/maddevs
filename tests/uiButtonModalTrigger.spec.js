import { mount } from '@vue/test-utils'
import UIButtonModalTrigger from '@/components/ui/UIButtonModalTrigger'

describe('ui button modal trigger', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(UIButtonModalTrigger, {
      propsData: {
        buttonInnerText: 'Some button name',
        modalWindowName: 'Some modal window name',
        isRed: true,
      },
      mocks: {
        $modal: {
          show: jest.fn(),
        },
        $nuxt: {
          $emit: jest.fn(),
        },
      },
    })
  })

  // ------ IMPORTANT ----- //
  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  // --------------------- //

  it('sets the correctly button name and contains new class --red', () => {
    const button = wrapper.find('.ui-button-modal-trigger')
    expect(button.text()).toBe('Some button name')
    expect(button.classes()).toContain('ui-button-modal-trigger--red')
  })
})
