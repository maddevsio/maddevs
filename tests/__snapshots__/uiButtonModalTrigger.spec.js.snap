// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui button modal trigger renders correctly 1`] = `
<button
  class="ui-button-modal-trigger ui-button-modal-trigger--red"
  modalwindowname="Some modal window name"
>
  
  Some button name

</button>
`;
